File: 251115/원점으로부터의 거리/distance-from-origin.py
class Point:
    def __init__(self, num, x, y):
        self.num = num
        self.x = x
        self.y = y

n = int(input())
points = []

num = 1
for _ in range(n):
    x, y = map(int, input().split())
    points.append(Point(num, x, y))
    num += 1

points.sort(key=lambda x: (abs(x.x) + abs(x.y)))

for p in points:
    print(p.num)

File: 260224/원점으로부터의 거리/distance-from-origin.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);

class Point {
    constructor(x, y, idx) {
        this.x = x;
        this.y = y;
        this.idx = idx;
    }
}

const compareDistance = (a, b) => {
    let distA = Math.abs(0 - a.x) + Math.abs(0 - a.y);
    let distB = Math.abs(0 - b.x) + Math.abs(0 - b.y);

    return distA - distB;
}

const points = [];
for (let i = 1; i < n + 1; i++) {
    let [x, y] = input[i].split(' ').map(Number);
    points[i - 1] = new Point(x, y, i);
}

points.sort(compareDistance);

for (let i = 0; i < n; i++) {
    console.log(`${points[i].idx}`);
}